merkletree.go

```diff
@@ -80,12 +93,13 @@ func lookUp(key string, str *SignedTreeRoot) (MerkleNode, []MerkleNode, error) {
 	}
 	switch nodePointer.(type) {
 	case *userLeafNode:
+		proof = append(proof, nodePointer.(*userLeafNode))
 		if nodePointer.(*userLeafNode).key == key {
-			proof = append(proof, nodePointer.(*userLeafNode))
 			return nodePointer.(*userLeafNode), proof, nil
-		} else {
-			// TODO: what should we return here?
 		}
+		// reached a different leaf with a matching prefix
+		// return nil and a auth path including the leaf node
+		return nil, proof, nil
```

Comment:
The verifier could see only exported contents through `LookUpProofNode` interface (https://github.com/coniks-sys/libmerkleprefixtree-go/blob/0cc51a16ed2f2a23023043ee3b72e7861f861657/node.go#L30). Currently, the exported content is the hash value of the node. I'm not sure what else should be exported but it could be extended in the future.